File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        
        // 1. 가로와 세로를 정렬
        for(int i=0;i<sizes.length;i++){
            for(int j=0;j<sizes[0].length;j++){
                if(sizes[i][0]>sizes[i][1]){
                    int tmp=sizes[i][0];
                    sizes[i][0]=sizes[i][1];
                    sizes[i][1]=tmp;
                }
            }
        }
        
        // 2. 가로와 세로의 최대값 구하기
        int wMax=0;
        int hMax=0;
        for(int[] n:sizes){
            wMax=Math.max(wMax,n[0]);
            hMax=Math.max(hMax,n[1]);
        }
        
        return wMax*hMax;
        
        
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    for size in sizes:
        size.sort(reverse=True)
        print(size)
    w,h=map(list,zip(*sizes))
    return max(w)*max(h)

# def solution(sizes):
#     # size들의 크기 정렬
#     for size in sizes:
#         size.sort(reverse=True)
#     # 가로 길이 리스트와 세로길이 리스트로 sizes들 언팩!!
#     w,h = map(list,zip(*sizes))
#     # 그중 최대끼리의 곱
#     return max(w) * max(h)